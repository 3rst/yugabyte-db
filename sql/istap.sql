\unset ECHO
\i test_setup.sql

SELECT plan(37);

/****************************************************************************/
-- Test is().
SELECT * FROM check_test( is(1, 1), true, 'is(1, 1)', '', '' );
SELECT * FROM check_test( is('x'::text, 'x'::text), true, 'is(''x'', ''x'')', '', '' );
SELECT * FROM check_test( is(1.1, 1.10), true, 'is(1.1, 1.10)', '', '' );
SELECT * FROM check_test( is(true, true), true, 'is(true, true)', '', '' );
SELECT * FROM check_test( is(false, false), true, 'is(false, false)', '', '' );
SELECT * FROM check_test( is(1, 1, 'foo'), true, 'is(1, 1, desc)', 'foo', '' );
SELECT * FROM check_test( is( 1, 2 ), false, 'is(1, 2)', '', '       have: 1
        want: 2');

/****************************************************************************/
-- Test isnt().
SELECT * FROM check_test( isnt(1, 2), true, 'isnt(1, 2)', '', '' );
SELECT * FROM check_test( isnt( 1, 1 ), false, 'isnt(1, 1)', '', '   1
      <>
    1' );

/****************************************************************************/
-- Try using variables.
\set foo '\'' waffle '\''
\set bar '\'' waffle '\''
SELECT is( :foo::text, :bar::text, 'is() should work with psql variables' );

/****************************************************************************/
-- Try using NULLs.

SELECT * FROM check_test(
    is( NULL::text, NULL::text, 'NULLs' ),
    true,
    'is(NULL, NULL)',
    'NULLs',
    ''
);

SELECT * FROM check_test(
    is( NULL::text, 'foo' ),
    false,
    'is(NULL, foo)',
    '',
    '       have: NULL
        want: foo'
);

SELECT * FROM check_test(
    is( 'foo', NULL::text ),
    false,
    'is(foo, NULL)',
    '',
    '       have: foo
        want: NULL'
);

/****************************************************************************/
-- Finish the tests and clean up.
SELECT * FROM finish();
ROLLBACK;
