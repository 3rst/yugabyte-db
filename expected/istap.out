\set ECHO
1..27
ok 1 - is() success
ok 2 - isa(1, 1) should work
ok 3 - is() success 2
ok 4 - is('x', 'x') should work
ok 5 - is() success 3
ok 6 - is(1.1, 1.10) should work
ok 7 - is() success 4
ok 8 - is(1.1, 1.10) should work
ok 9 - is() success 5
ok 10 - is(true, true) should work
ok 11 - is() success 6
ok 12 - is(false, false) should work
ok 13 - is() success 7
ok 14 - is(1, 1, 'foo') should work
ok 15 - is() failure
ok 16 - is(1, 2) should work
ok 17 - isnt() success
ok 18 - isnt(1, 2) should work
ok 19 - isnt() failure
ok 20 - is(1, 2) should work
ok 21 - is() should work with psql variables
ok 22 - is(NULL, NULL) success
ok 23 - is(NULL, NULL) should pass
ok 24 - is(NULL, foo) failure
ok 25 - is(NULL, foo) should fail
ok 26 - is(foo, NULL) failure
ok 27 - is(foo, NULL) should fail
